client/homebrew/homebrew.jsx

```diff
@@ -12,6 +13,21 @@ const NewPage = require('./pages/newPage/newPage.jsx');
 //const ErrorPage = require('./pages/errorPage/errorPage.jsx');
 const PrintPage = require('./pages/printPage/printPage.jsx');
 
+const WithRoute = (props)=>{ //(Element, additionalPropsFn)=>{
+	const params = useParams();
+	const location = useLocation();
+	let additionalProps = {};
+	if(props.additionalPropsFn) {
+		additionalProps = props.additionalPropsFn({ props, params, location });
+	}
```

Comment:
Looks like react-router-dom v6' now has a built-in way to handle URL query parameters using the new [useSearchParams](https://reactrouter.com/docs/en/v6/hooks/use-search-params) hook. I think that means we can probably get rid of `additionalPropsFn` to handle that and just do a one-liner `const queryParams = useSearchParams();`

This also means we can get rid of the `query-string` dependency.